### DISCUSSION TITLE
a way to parse URLSearchParams

### DISCUSSION THREAD

**Discussion by JacobWeisenburger** [colinhacks/zod#1770] [Show and tell]
https://gist.github.com/JacobWeisenburger/9256eae415f6b0a04b718d633266a4e0

Please let me know what you think of this. Always glad to get feedback, positive or negative.

**Comment by maxArturo** [comment]
I can see myself reaching for this if I already use Zod and I don't need the usual heavy-handed npm packages _just_ for doing this. Thanks!

**Reply by JacobWeisenburger** [reply]
do you have an example of a "heavy-handed npm package"? I'm not sure I know what you mean

**Comment by JacobWeisenburger** [comment]
I have also implemented this in [zod_utilz](https://github.com/JacobWeisenburger/zod_utilz)
https://github.com/JacobWeisenburger/zod_utilz#useurlsearchparams

Always open to suggestions. Feel free to contribute an issue or PR.
